src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -50,23 +53,32 @@ protected override void ExecuteCore()
             IEnumerable<string> privateAssetsPackageIds = PackageReferenceConverter.GetPackageIds(PrivateAssetsPackageReferences);
             IPackageResolver packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
 
-            LockFile filterLockFile = null;
-            if (!string.IsNullOrEmpty(FilterProjectAssetsFile))
+            HashSet<PackageInfo> packagestoBeFiltered = null;
+            if (!string.IsNullOrEmpty(FilterProjectFiles))
             {
-                filterLockFile = lockFileCache.GetLockFile(FilterProjectAssetsFile);
+
+                packagestoBeFiltered = CacheArtifactParser.Parse(FilterProjectFiles);
+
+                Log.LogMessage(MessageImportance.Low, string.Format(CultureInfo.CurrentCulture, Strings.ParsingFiles, FilterProjectFiles));
```

Comment:
It seems like this should go above actually calling the Parser, since the string is "ParsingFiles".